jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildDockerTask.java

```diff
@@ -16,18 +16,130 @@
 
 package com.google.cloud.tools.jib.gradle;
 
+import com.google.api.client.http.HttpTransport;
+import com.google.cloud.tools.jib.builder.BuildConfiguration;
+import com.google.cloud.tools.jib.frontend.BuildDockerStepsRunner;
+import com.google.cloud.tools.jib.frontend.BuildImageStepsExecutionException;
+import com.google.cloud.tools.jib.frontend.CacheDirectoryCreationException;
+import com.google.cloud.tools.jib.http.Authorization;
+import com.google.cloud.tools.jib.http.Authorizations;
+import com.google.cloud.tools.jib.image.ImageReference;
+import com.google.cloud.tools.jib.image.InvalidImageReferenceException;
+import com.google.cloud.tools.jib.registry.RegistryClient;
+import com.google.cloud.tools.jib.registry.credentials.RegistryCredentials;
+import com.google.common.base.Preconditions;
+import java.nio.file.Path;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 import javax.annotation.Nullable;
 import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.tasks.Nested;
 import org.gradle.api.tasks.TaskAction;
 
 /** Builds a container image and exports to the default Docker daemon. */
 public class BuildDockerTask extends DefaultTask {
 
+  /** Directory name for the cache. The directory will be relative to the build output directory. */
+  private static final String CACHE_DIRECTORY_NAME = "jib-cache";
```

Comment:
We'd probably want this to be shared and the same with `BuildImageTask`. Maybe `ProjectProperties` would be a good place for it, and the tasks would call like `ProjectProperties#getCacheDirectory`.

Reply:
Do you want me to do this now or hold off on this until last bullet of #262?

Reply:
Can hold off but maybe add a TODO here or at the top of the class